darts/tests/models/forecasting/test_historical_forecasts.py

```diff
@@ -1697,124 +1740,149 @@ def test_torch_auto_start_with_past_cov(self, model_config):
-            theorical_forecast_length = (
-                self.ts_val_length
-                - (bounds[0] + bounds[1])  # train sample length
-                - (
-                    forecast_hrz - 1
-                )  # (horizon - 1): if entire horizon is available, we can predict 1,
-                - 7  # future covs start 7 after target (more than past covs) -> shift
-                - 2  # future covs in output chunk -> difference between horizon=10 and output_chunk_length=12
-            )
-            assert len(forecasts[0]) == theorical_forecast_length, (
-                f"Model {model_cls} does not return the right number of historical forecasts in case "
-                f"of retrain=True and overlap_end=False and no past_covariates and future_covariates "
-                f"with different start. "
-                f"Expected {theorical_forecast_length}, got {len(forecasts[0])}"
-            )
-            theorical_forecast_length = (
-                self.ts_val_length
-                - (bounds[0] + bounds[1])  # train sample length
-                - (forecast_hrz - 1)  # if entire horizon is available, we can predict 1
-                - 0  # all covs start at the same time as target -> no shift
-                - 2  # future covs in output chunk -> difference between horizon=10 and output_chunk_length=12
-            )
-            assert len(forecasts[1]) == theorical_forecast_length, (
-                f"Model {model_cls} does not return the right number of historical forecasts in case "
-                f"of retrain=True and overlap_end=False and no past_covariates and future_covariates "
-                f"with different start. "
-                f"Expected {theorical_forecast_length}, got {len(forecasts[1])}"
+        model = model_cls(
+            random_state=0,
+            **kwargs,
+        )
+
+        if not model.supports_future_covariates:
+            with pytest.raises(ValueError) as err:
+                model.fit(self.ts_pass_train, future_covariates=self.ts_fut_cov_train)
+            assert str(err.value).startswith(
+                "The model does not support `future_covariates`"
             )
+            return
+
+        model.fit(self.ts_pass_train, future_covariates=self.ts_fut_cov_train)
+
+        # Only fut covariate
+        forecasts = model.historical_forecasts(
+            series=[self.ts_pass_val, self.ts_pass_val],
+            future_covariates=[
+                self.ts_fut_cov_valid_7_aft_start,
+                self.ts_fut_cov_valid_16_bef_start,
+            ],
+            forecast_horizon=forecast_hrz,
+            stride=1,
+            retrain=True,
+            overlap_end=False,
+        )
+
+        assert (
+            len(forecasts) == 2
+        ), f"Model {model_cls} did not return a list of historical forecasts"
+
+        icl, ocl = bounds
+        theorical_forecast_length = (
+            self.ts_val_length
+            - (icl + ocl)  # train sample length
```

Comment:
Wouldn't it be better to have `min_train_series_length` instead?

Reply:
As discussed, will be part of another PR that fixes the min_train_series_length overall